Providers/RustErrors.swift

```diff
@@ -0,0 +1,74 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import Foundation
+import MozillaAppServices
+
+public func initializeRustErrors(crashManager: CrashManager) {
+    setApplicationErrorReporter(errorReporter: FirefoxIOSErrorReporter(crashManager: crashManager))
+}
+
+/// The `AppServicesErrorReport` class (with its inheritance from `RustCrashReport`) exists
+/// to distinguish native Sentry reports from reports originating in A-S
+private class AppServicesErrorReport: Error, RustCrashReport {
+    var typeName: String
+    var message: String
+
+    init(typeName: String, message: String) {
+        self.typeName = typeName
+        self.message = message
+    }
+}
+
+/// The `FirefoxIOSErrorReporter` class contains the callbacks A-S uses to report Sentry errors and
+/// breadcrumbs. These functions are not intended to be explicitly called in this repo.
+private class FirefoxIOSErrorReporter: ApplicationErrorReporter {
+    var crashManager: CrashManager
+
+    init(crashManager: CrashManager) {
+        self.crashManager = crashManager
+    }
+
+    func reportError(typeName: String, message: String) {
+        crashManager.captureError(error: AppServicesErrorReport(typeName: typeName, message: message))
+    }
+
+    func reportBreadcrumb(message: String, module: String, line: UInt32, column: UInt32) {
+        crashManager.addBreadcrumb(message: "\(module)[\(line)]: \(message)",
+                                   category: LoggerCategory.sync,
+                                   level: LoggerLevel.info)
+    }
+}
+
+/// The `ForwardOnLog` class exists to support the rust-log-forwarder `setLogger` function.
+internal class ForwardOnLog: AppServicesLogger {
+    var logger: Logger
+
+    init(logger: Logger) {
+        self.logger = logger
+    }
+
+    func log(record: Record) {
+        self.logger.rustLog(record.message,
+                            level: rustLevelToLoggerLevel(level: record.level),
+                            category: LoggerCategory.sync,
+                            extra: ["target": record.target])
```

Comment:
Thanks for this, @OrlaM. My original thought was I wanted reduce the amount of breadcrumbs that were being reported since we will be reporting breadcrumbs from A-S. However I think this is premature on my part at minimum, and possibly unnecessary or even undesirable.

I'm going to remove this logic. I can address any inundation of A-S breadcrumbs if/as they arise.

Reply:
Yeah it should be fine for now. I'm going to be looking into a better long term solution for consuming logs in the near future, since either having users manually send us their logs or the small snippet of logs we get from Sentry in the breadcrumbs aren't ideal. I'll keep you posted when we start to make a move on that.